src/lib.rs

```diff
@@ -851,65 +853,83 @@ impl LightningNode {
+            filters: Some(vec![
+                PaymentTypeFilter::Sent,
+                PaymentTypeFilter::Received,
+                PaymentTypeFilter::ClosedChannel,
+            ]),
             from_timestamp: None,
             to_timestamp: None,
             include_failures: Some(true),
-            limit: Some(number_of_completed_payments + LEEWAY_FOR_PENDING_OUTGOING_PAYMENTS),
+            limit: Some(number_of_completed_movements + LEEWAY_FOR_PENDING_PAYMENTS),
             offset: None,
         };
-        let breez_payments = self
+        let breez_movements = self
             .rt
             .handle()
             .block_on(self.sdk.list_payments(list_payments_request))
             .map_to_runtime_error(RuntimeErrorCode::NodeUnavailable, "Failed to list payments")?
             .into_iter()
-            .map(|p| self.payment_from_breez_payment(p))
-            .filter_map(filter_out_and_log_corrupted_payments)
+            .map(|p| self.movement_from_breez_payment(p))
+            .filter_map(filter_out_and_log_corrupted_movements)
             .collect::<Vec<_>>();
 
         // Query created invoices, filter out ones which are in the breez db.
-        let breez_payment_hashes: HashSet<_> = breez_payments
+        let breez_payment_hashes: HashSet<_> = breez_movements
             .iter()
-            .map(|p| p.invoice_details.payment_hash.as_ref())
+            .filter_map(|m| match m {
+                Movement::Payment { payment } => {
+                    Some(payment.invoice_details.payment_hash.as_ref())
+                }
+                Movement::ChannelClose { .. } => None,
+            })
             .collect();
         let created_invoices = self
             .data_store
             .lock_unwrap()
-            .retrieve_created_invoices(number_of_completed_payments)?;
-        let created_invoices = created_invoices
+            .retrieve_created_invoices(number_of_completed_movements)?;
+        let created_invoice_movements = created_invoices
             .into_iter()
             .filter(|i| !breez_payment_hashes.contains(i.hash.as_str()))
             .map(|i| self.payment_from_created_invoice(&i))
             .filter_map(filter_out_and_log_corrupted_payments)
+            .map(|p| Movement::Payment { payment: p })
             .collect::<Vec<_>>();
 
         // Split by state.
-        let mut pending_payments = Vec::new();
-        let mut completed_payments = Vec::new();
-        for payment in breez_payments.into_iter().chain(created_invoices) {
-            match payment.payment_state {
-                PaymentState::Created | PaymentState::Retried => pending_payments.push(payment),
-                _ => completed_payments.push(payment),
+        let mut pending_movements = Vec::new();
+        let mut completed_movements = Vec::new();
+        for movement in breez_movements.into_iter().chain(created_invoice_movements) {
```

Comment:
As you can tell from my previous comment, I don't find all this code to be very understandable. I wonder whether `created_invoice_movements` can ever end up in `completed_movmements`?

Reply:
`created_invoice_movements` (now called `unpaid_invoice_movements`) can end up in `completed_movements` in case the invoice has expired.

Reply:
Oh okay, intersting ... Thanks for elaborating